### ISSUE TITLE
Author *.qml for Ecoregion layers and test in QRAVE

### ISSUE THREAD

**Issue by Tanmanfalcon** [Riverscapes/RiverscapesXML#152]

**Comment by Tanmanfalcon** [comment]
![qrave](https://user-images.githubusercontent.com/79670857/126227200-9944eff1-70bc-4a1f-8b09-c363baca1b3d.PNG)
While attempting to copy symbology from ESRI to QGIS, I realized that the ecoregions and the vegetation layers have well over a thousand different symbology colors for the thousands of different attributes. Could there be a way to copy symbology from ESRI more effieciently than manually editing symbology?

**Comment by wally-mac** [comment]
@Cashe93 do you have any insights for @Tanmanfalcon?  I think we need to find a more time efficient solution.   What about just using the "canned" symbology that comes from the EPA?

**Comment by joewheaton** [comment]
> @Cashe93 do you have any insights for @Tanmanfalcon? I think we need to find a more time efficient solution. What about just using the "canned" symbology that comes from the EPA?

I don't think we have a more time efficient solution @wally-mac unless someone knows how to cover ecoregions from `*.lyr` to `*.qml`.  Might just be a slow tedious, suck it up and plough through I'm afraid.

**Comment by wally-mac** [comment]
@Tanmanfalcon it looks like there is a QGIS Pugin to convert ArcGIS .lyr files to QGIS .qml styles.  Based on this [Youtube video](https://www.youtube.com/watch?v=FEGso40qrQc) the SLYR plugin (community version) works by converting esri .style databases to QGIS .xml styles. The video also says that the full version will convert .lyr files to QGIS .qml.  Please have a look and let me know if this might be a solution.

**Comment by joewheaton** [comment]
If that works, that is huge @wally-mac and all the @Riverscapes/riverscape-project-curators should be interested in this. Please report back what you find if trying this!

**Comment by Cashe93** [comment]
@wally-mac, I'll give this a go with some BRAT symbology.

**Comment by Cashe93** [comment]
@wally-mac and @joewheaton, that tool appears to work. Looks like it will definitely save @Riverscapes/riverscape-project-curators some time when authoring symbology. The video that @wally-mac posted is a very straightforward tutorial.

**Comment by Cashe93** [comment]
> @wally-mac and @joewheaton, that tool appears to work. Looks like it will definitely save @Riverscapes/riverscape-project-curators some time when authoring symbology. The video that @wally-mac posted is a very straightforward tutorial.

@Riverscapes/riverscape-project-curators, my apologies, I just realized that this tool may not work exactly how we want it to since it only applies the symbology that's relevant to whatever project you are working on. So when you export the layer as a .qml, it will likely exclude some categories. I will look into some possible workarounds and keep you posted. There is a tool that directly converts .lyr to .qml but it requires a SLYR license.

**Comment by wally-mac** [comment]
@Cashe93 please look into how much a SLYR license is.  Depending on the cost it might be worth the investment if it does what it claims it will do -- convert .lyr to .qml.  Let them know we are academic because there is typically a discount.

**Comment by Cashe93** [comment]
@wally-mac, will do.

**Comment by joewheaton** [comment]
@Cashe93 can you share here for others what you guys have learned about SLYR?

**Comment by Cashe93** [comment]
Sure thing. @Riverscapes/riverscape-project-curators, there is a tool within the licensed version of SLYR that allows us to convert esri LYR files into qgis compatible files. We were able to send a couple of example files for a test run, and it looks like the outputs come back as QLR files. These files are similar to QML however, they are linked to a dataset while QML files just contain style information. Once a QLR is created, all you have to do is export it as a QML. We are in the process of purchasing a license that everybody can use. Once we have this I'll put together a short video on how to use the tool. In a nutshell, this tool should eliminate the need to manually copy over symbology. @Tanmanfalcon or @wally-mac, feel free to add anything else.

**Comment by lauren-herbine** [comment]
That's awesome! Thanks for looking into this @Cashe93. Looking forward to your video

**Comment by joewheaton** [comment]
Thanks @Cashe93, that will go in RAVE website: https://github.com/Riverscapes/RaveAddIn/issues/140

**Comment by wally-mac** [comment]
Update: I just purchased the SLYR software.

**Comment by Cashe93** [comment]
Hey all, here are the installation instructions for SLYR. 

Downloading and installing the SLYR plugin for QGIS

Important: Before installing the licensed version of SLYR, you must first uninstall the community version (if installed) and restart QGIS

1. Download the installer from https://usuwatershed:JloiY%24%29o%7Bskp@north-road.com/qgis_plugins/2872eb8e-bc44-46ac-9f11-f124c0ce4915/slyr_install.zip

2. Unzip the downloaded file, and then drag the extracted "install_slyr_qgis.py" over a running QGIS install. The script will add a connection to the private plugins repository, and install the SLYR plugin for you.

3. After the plugin is installed, the SLYR options dialog will open. You must enter your unique license key at this screen, exactly as it appears at the end of this email.

To use the plugin, simply drag and drop ESRI lyr, mxd, style, pmf or sxd files onto the main QGIS window. Alternatively, use the QGIS browser panel to locate lyr, mxd or style files and right click these for the available conversion options. The plugin also adds a new "SLYR" group to the Processing toolbox which contains numerous tools for converting and managing ESRI documents.

**Comment by Cashe93** [comment]
@Riverscapes/riverscape-project-curators, I've downloaded and installed the SLYR toolkit and everything seems to work great. The tool we will likely be utilizing the most is the LYR to QML tool which is very easy to use; just drag the desired LYR file as an input and the tool will output the corresponding QML (no exporting necessary). Since it's pretty straightforward, I don't think it's necessary that I make a video, but feel free to reach out to me if you have any questions. 

![image](https://user-images.githubusercontent.com/49880208/127924968-35be04bf-e52b-4818-9918-71981da969b7.png)

![image](https://user-images.githubusercontent.com/49880208/127925033-df983c18-fe30-4298-9cfd-309ed0af0bd8.png)

**Comment by lauren-herbine** [comment]
Hey @Cashe93 quick q- just to make sure, SLYR only works esri->Q, correct? So for efficiency's sake, it would make most sense to start all curation projects with Arc, and then use this tool to auto-create the corresponding .qml's?

**Comment by Cashe93** [comment]
@lauren-herbine, I believe that is correct. SLYR appears to have a handful of tools that converts LYR to Q compatible layers, but not vice versa. So yes, I would author symbology within Arc and then convert. I will send an email to the SLYR folks just to confirm this.

**Comment by lauren-herbine** [comment]
@Cashe93 great, thanks for the confirmation!
